MSOA data

ShahrajKarim opened the pull request:
- Added code to call data at MSOA level
- Fixed issue for OA level child count which previously did not sum across ages.

ShahrajKarim committed:
- Added code to call data at MSOA level
- Fixed issue for OA level child count which previously did not sum across ages.

lasse-meixner reviewed (commented):
I haven't tried running but looks practical 👍 - for the future, if all these calls are the same except for the geography type argument, it'd be nicer to have them all wrapped in a single function with a location argument than have everything "twice".

ShahrajKarim committed:
API script now loops over OA, MSOA and LSOA.
Removed data on child count, now replaced it with households with dependent children under age of 9 and called it eligibility

ShahrajKarim commented:
Looped the API call now. Let me know if I need to document things more clearly.

lasse-meixner commented on scripts/leeds_census_2021.R:
```diff
@@ -1,138 +1,132 @@
+library(googledrive)
+
+# Define function to get census data
+get_census_data <- function(geography, geo_col) {
```
What do these two arguments refer to? 🤔

ShahrajKarim replied:
geography is the "geography code" that nomisr uses to identify the level at which the variable is being called in at e.g. geography = "TYPE150" means that we're calling in variables at the OA level

geo_col is just an identifier for oa21cd, lsoa21cd or msoa21cd and I used this to left_join the data by the same oa/lsoa/msoa

lasse-meixner replied:
👍 add this in a docstring :)

lasse-meixner commented on scripts/leeds_census_2021.R:
```diff
@@ -1,138 +1,132 @@
+    dep_child_category = 3          # Rename third column
+  )
+
+lookup_data <- read.csv("raw_data/geography_lookup_Dec_2021.csv") %>%
+  rename_with(tolower) %>%  # Convert column names to lowercase
+  select(ends_with("cd"))   # Keep only columns ending with "cd"
```
I think you already load this object above?

ShahrajKarim replied:
Yes you're right, that's a good spot! I'll remove this.

lasse-meixner commented on scripts/leeds_census_2021.R:
```diff
@@ -1,138 +1,132 @@
+geo_levels <- c("oa21cd", "lsoa21cd", "msoa21cd")
+
+# Loop over each geography level and store results in a list
+results <- lapply(geo_levels, function(geo) {
+  if (geo %in% colnames(dep_child_age_data)) {  # Check if the column exists
+    dep_child_age_data %>%
+      group_by(across(all_of(geo))) %>%
+      summarise(
+        total_number_of_households = sum(Observation, na.rm = TRUE),
+        eligible = sum(Observation[dep_child_category %in% c(1, 2)], na.rm = TRUE),
+        share_of_eligible = 100 * eligible / total_number_of_households,
+        .groups = "drop"  # Ensure ungrouping after summarising
+      ) %>%
+      select(all_of(geo), total_number_of_households, eligible, share_of_eligible)  # Keep only required columns
+  } else {
+    message(paste("Skipping:", geo, "not found in dataset"))
+    return(NULL)  # Return NULL if the column does not exist
+  }
+})
```
This is for creating new aggregate statistics for each geographic unit (e.g. LSOA), for each level of aggregation in geo_levels?

ShahrajKarim replied:
Yes - this is specifically for households with dependent children.

lasse-meixner replied:
Could we, in addition, also have the more granular info, the ratio of the [0-4] and [5-9]?

ShahrajKarim replied:
Sure, I just committed some more changes

lasse-meixner reviewed (approved):
Just a few questions, otherwise LGTM.

ShahrajKarim committed:
Added comments in code to make it easier to follow

ShahrajKarim committed:
Added ratios for [0,4] and [5,9]
